wireapp/wire-ios-mono
# fix: passcode during call - reuse `LAContext` [task][WPB-7111]

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7111" title="WPB-7111" target="_blank"><img alt="Bug" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10803?size=medium" />WPB-7111</a>  [iOS] C builds asks for passcode during call
  </td></table>
  <br />
 






















----
#### PR Submission Checklist for internal contributors

- The **PR Title**

- The **PR Description**
----

> [!NOTE]
> Merges into feature branch #1188

This also solves the ticket https://wearezeta.atlassian.net/browse/WPB-87 ✅

# What's new in this PR?

### Issues

So now `LAContext` is accessed through the `AuthenticationContext` which makes sure we that we only have one instance around and the instance is reused in order not to prompt the user to enter the passcode. The `LAContext` object is released whenever the app goes into the background. Previously an LAContext was created implicitly when calling certain methods.

This change doesn't change any properties of how we can access the database since the `LAContext` is still only used to unlock the database keys and after that we hold on to the database keys and this hasn't changed.

This solves the existing issue, that the user had to enter the passcode two times on app startup - one time for the custom AppLock and one time for the access to the keychain.

Unfortunately the existing issue [WPB-7111] is not solved and the prompt still appears during calls. I will make another follow up to address this issue.

### Testing

#### Test Coverage (Optional)


#### How to Test

- Build app that enforces encryption at rest:
  - In `session_manager.json` change the value to `"encryptionAtRestEnabledByDefault": true`
  - In `Security-Flags.xcconfig` change the value to `FORCE_ENCRYPTION_AT_REST_ENABLED=1`
- Login with a user
- The prompt for authentication should appear only once

----
#### PR Post Submission Checklist for internal contributors (Optional)

----
#### PR Post Merge Checklist for internal contributors

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.


[WPB-7111]: https://wearezeta.atlassian.net/browse/WPB-7111?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
